File: Python/programmers/etc/level2/행렬의_곱셈.py
def solution(arr1, arr2):
    n, m, r = len(arr1), len(arr1[0]), len(arr2[0])
    answer = [[0]*r for _ in range(n)]
    for i in range(n):
        for j in range(m):
            for k in range(r):
                answer[i][k] += arr1[i][j]*arr2[j][k]
    return answer


arr1, arr2 = [[1, 4], [3, 2], [4, 1]], [[3, 3], [3, 3]]
print(solution(arr1, arr2)) # [[15, 15], [15, 15], [15, 15]]

arr1, arr2 = [[2, 3, 2], [4, 2, 4], [3, 1, 4]], [[5, 4, 3], [2, 4, 1], [3, 1, 1]]
print(solution(arr1, arr2)) # [[22, 22, 11], [36, 28, 18], [29, 20, 14]]

File: Java/programmers/etc/level2/행렬의곱셈.java
package CodeTest.Java.programmers.etc.level2;

public class 행렬의곱셈 {
    public int[][] solution(int[][] arr1, int[][] arr2) {
        int n = arr1.length, m = arr1[0].length, r = arr2[0].length;

        int[][] answer = new int[n][r];

        for (int i=0; i<n; i++) {
            for (int j=0; j<m; j++) {
                for (int k=0; k<r; k++) {
                    answer[i][k] += arr1[i][j]*arr2[j][k];
                }
            }
        }

        return answer;
    }
}
